api/src/main/java/org/eclipse/microprofile/metrics/MetricID.java

```diff
@@ -0,0 +1,236 @@
+ * may contain any UTF-8 encoded character. Global tags can be set by passing
+ * the list of tags in an environment variable {@code MP_METRICS_TAGS}. Tag values set
+ * through `MP_METRICS_TAGS` MUST escape equal symbols `=` and commas `,` with a backslash `\`
+ *
+ * For example, the following can be used to set the global tags:
+ *
+ * <pre>
+ * <code>
+ *      export MP_METRICS_TAGS=app=shop,tier=integration,special=deli\=ver\,y
+ * </code>
+ * </pre>
+ *
+ * </li>
+ * </ul>
+ */
+public class MetricID {
+
+    public static final String GLOBAL_TAGS_VARIABLE = "MP_METRICS_TAGS";
+
+    /**
+     * Name of the metric.
+     * <p>
+     * A required field which holds the name of the metric object.
+     * </p>
+     */
+    private final String name;
+
+    /**
+     * Tags of the metric. Augmented by global tags.
+     * <p>
+     * An optional field which holds the tags of the metric object which can be
+     * augmented by global tags.
+     * </p>
+     */
+    private final Map<String, String> tags = new HashMap<String, String>();
+
+    /**
+     * Constructs a MetricID with the given metric name and no tags.
+     * If global tags are available then they will be appended to this MetricID.
+     *
+     * @param name the name of the metric
+     */
+    public MetricID(String name) {
+        this(name, null);
+    }
+
+    /**
+     * Constructs a MetricID with the given metric name and {@link Tag}s.
+     * If global tags are available then they will be appended to this MetricID
+     *
+     * @param name the name of the metric
+     * @param tags the tags associated with this metric
+     */
+    public MetricID(String name, Tag... tags) {
+        this.name = name;
+        String globalTagsFromEnv = System.getenv(GLOBAL_TAGS_VARIABLE);
+        parseGlobalTags(globalTagsFromEnv);
+        addTags(tags);
+    }
+
+    /**
+     * Returns the Metric name associated with this MetricID
+     *
+     * @return the Metric name associated with this MetricID
+     */
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * Returns the underlying HashMap containing the tags.
+     *
+     * @return a {@link Map} of tags
+     */
+    public Map<String, String> getTags() {
+        return Collections.unmodifiableMap(tags);
+    }
+
+    /**
+     * Gets the list of tags as a single String in the format 'key="value",key2="value2",...'
+     *
+     * @return a String containing the tags
+     */
+    public String getTagsAsString() {
+        return this.tags.entrySet().stream().map(e -> e.getKey() + "=\"" + e.getValue() + "\"").collect(Collectors.joining(","));
+    }
+
+    /**
+     * Gets the list of tags as a list of {@link Tag} objects
+     *
+     * @return a a list of Tag objects
+     */
+    public List<Tag> getTagsAsList() {
+        List<Tag> list = new ArrayList<Tag>();
+        this.tags.entrySet().stream().forEach(entry -> list.add(new Tag(entry.getKey(), entry.getValue())));
```

Comment:
`this.tags.forEach((key, value) -> list.add(new Tag(key, value)));` is shorter and doesn't create a stream (which is slow)